packages/scanner/zxing/android/src/main/kotlin/org/openfoodfacts/scanner_zxing/ScannerZxingPlugin.kt

```diff
@@ -0,0 +1,37 @@
+package org.openfoodfacts.scanner_zxing
+
+import androidx.annotation.NonNull
+
+import io.flutter.embedding.engine.plugins.FlutterPlugin
+import io.flutter.plugin.common.MethodCall
+import io.flutter.plugin.common.MethodChannel
+import io.flutter.plugin.common.MethodChannel.MethodCallHandler
+import io.flutter.plugin.common.MethodChannel.Result
+
+/** ScannerZxingPlugin */
+class ScannerZxingPlugin : FlutterPlugin, MethodCallHandler {
+
+    private lateinit var channel: MethodChannel
+
+    override fun onAttachedToEngine(@NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
+        channel = MethodChannel(flutterPluginBinding.binaryMessenger, "openfoodfacts/zxing")
+        channel.setMethodCallHandler(this)
+    }
+
+    override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
+        if (call.method == "scanFile") {
+            result.success(
+                ZXingUtils.extractBarcodeFromImage(
+                    call.argument("path") as String?,
+                    call.argument("orientation") as Int?,
```

Comment:
If any of these two are mandatory, I would remove the nullable and make it crash if we pass something wrong. It could be subdle to find issues related to the scanner not working just to find out we're passing null somewhere

Reply:
Actually, the check is done in the calling method, hence the nullable accepted here